tools/app/app/components/MetadataView.tsx

```diff
@@ -27,19 +28,71 @@ import {
   FieldTypeDefinition,
   MetadataValue,
 } from '../lib/definitions';
-import EditableField from './EditableField';
+import EditableField, { EditableFieldProps } from './EditableField';
+
+interface FieldItemProps {
+  expanded?: boolean;
+  control?: Control;
+  defaultValue: MetadataValue | null;
+  editableFieldProps: Omit<Omit<EditableFieldProps, 'onChange'>, 'value'>;
+  name: string;
+  handleChange?: (
+    e: any,
+    onChange: (arg0: MetadataValue) => void,
+    dataType: DataType,
+  ) => void;
+}
+
+function FieldItem({
+  expanded,
+  control,
+  name,
+  defaultValue,
+  editableFieldProps,
+  handleChange,
+}: FieldItemProps) {
+  return (
+    <Accordion expanded={expanded}>
+      <AccordionDetails>
+        {control ? (
+          <Controller
+            name={name}
+            control={control}
+            defaultValue={defaultValue}
+            render={({ field: { value, onChange } }: any) => {
+              return (
+                <EditableField
+                  value={value}
+                  onChange={(e) => {
+                    if (handleChange)
+                      handleChange(e, onChange, editableFieldProps.dataType);
+                  }}
+                  {...editableFieldProps}
+                />
+              );
+            }}
+          />
+        ) : (
+          <EditableField value={defaultValue} {...editableFieldProps} />
+        )}
+      </AccordionDetails>
+    </Accordion>
+  );
+}
 
 export interface MetadataViewProps {
   initialExpanded?: boolean;
   editMode?: boolean;
   metadata?: CardMetadata;
```

Comment:
Not directly related to this PR, but I guess `metadata` could be mandatory in app. That would simplify some of the handling. I think app always retrieve metadata with the card information.

Reply:
Yes, that would be a good idea